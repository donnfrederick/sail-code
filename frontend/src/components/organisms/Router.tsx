import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
