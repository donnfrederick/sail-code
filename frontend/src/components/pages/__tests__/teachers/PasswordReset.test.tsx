import Router from 'components/organisms/Router'
import TeachersPasswordReset from 'components/pages/teachers/PasswordReset'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <TeachersPasswordReset />', () => {
  const tree = renderer
    .create(
      <Router>
        <TeachersPasswordReset />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
