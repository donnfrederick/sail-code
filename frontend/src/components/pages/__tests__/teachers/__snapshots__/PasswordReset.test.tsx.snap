// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <TeachersPasswordReset /> 1`] = `
<div
  className="sc-kkGfuU kBHvTG"
>
  <div
    className="sc-jKJlTe hHLXRu"
  >
    <div
      className="sc-eNQAEJ cuxGLA"
    >
      登録したメールアドレスを
入力してください
    </div>
    <div
      className="sc-cSHVUG dJUEwk"
    >
      <div
        className="sc-kAzzGY kLtOd"
        data-error={false}
      >
        メールアドレス
      </div>
      <input
        className="sc-chPdSV hkBQDR"
        data-error={false}
        defaultValue=""
        maxLength={254}
        name="mail"
        onInput={[Function]}
        placeholder="sail@example.com"
      />
    </div>
    <div
      className="sc-kEYyzF gSLsJy"
    >
      <button
        className="sc-gZMcBi dNKJEZ"
        data-active={true}
        fontSize={40}
        height={112}
        width={304}
      >
        <a
          className="sc-fjdhpX fRklgf"
          href="/teachers/signin"
          onClick={[Function]}
        >
          戻る
        </a>
      </button>
      <button
        className="sc-gqjmRU bhAvEZ"
        data-active={false}
        fontSize={40}
        height={112}
        onClick={[Function]}
        width={304}
      >
        <div
          className="sc-jzJRlG dIrfbV"
        >
          送信
        </div>
      </button>
    </div>
  </div>
  <div
    className="sc-EHOje itwipb"
  >
    <div
      className="sc-bZQynM bZKBNj"
    >
      パスワードの再設定
    </div>
    <a
      className="sc-gzVnrw jTRbiO"
      href="/teachers"
      onClick={[Function]}
    >
      最初に戻る
    </a>
  </div>
</div>
`;
